### ISSUE TITLE
"Lock the scale to use magnifier to zoom in or out" is not working properly

### ISSUE THREAD

**Issue by SeqLaz** [qgis/QGIS#56803]
### What is the bug or the crash?

When utilizing the "Lock the scale to use magnifier to zoom in or out" feature, its functionality appears inconsistent. At times, it fails to activate despite the symbols remaining fixed at a specific scale. Conversely, on other occasions, it activated, yet the symbols alter their scale.

![](https://github.com/qgis/QGIS/assets/77129293/3a53ec21-53b3-42f1-bc66-5e0d23dde2e2)

![](https://github.com/qgis/QGIS/assets/77129293/58887732-4667-444e-a61a-3956f36db528)

![image](https://github.com/qgis/QGIS/assets/77129293/1fe26415-7f1a-470c-a24b-35f6e5410cf3)

### Steps to reproduce the issue

1. When activating the "Lock the scale to use magnifier to zoom in or out" feature and subsequently using the scale next to it to zoom in or out, the symbols on the canvas may either change or remain fixed at their original scale.

2. Conversely, upon deactivating the "Lock the scale to use magnifier to zoom in or out" feature, the scale may either change or remain the same.

3. Unfortunately, only restarting QGIS rectify the problem.

### Versions

Tested on Windows 11 and Ubuntu 22.04

QGIS version 3.34.4 Prizren
QGIS version 3.36.0-Maidenhead

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by frspp** [comment]
I have not noted this behauviour, still valid?

**Comment by SeqLaz** [comment]
> I have not noted this behauviour, still valid?

Yes, still behaving strange, it happen when activate and deactivate the "Lock the scale to use magnifier to zoom in or out" button.

https://github.com/user-attachments/assets/fe13dc14-3c55-4f6e-b18a-0c9f6614ee64

**Comment by North35** [comment]
I’m not sure if this is the same issue, but I think the behavior of scale locking is unintuitive. After unlocking the scale, the magnifier value should return to the default value of 100%. Instead, it remains at whatever value it had before the scale was unlocked. As a result, the user may wonder why, at the scale they were just using, the objects are displayed completely differently than just minutes before. This can be seen in the third screenshot, where the scale is unlocked but the magnifier shows 800%.

**Comment by frspp** [comment]
- @North35 my first feeling is opposite: scale/zoom should not automatically change to another value when unlocked.

**Comment by North35** [comment]
I think the scale should always have priority, and locking or unlocking it shouldn’t affect how the user perceives it. When you unlock the scale, you want to see objects at the scale that the program is displaying at that moment. But everyone has their own preferences, we can agree to disagree.
